cpp/src/arrow/testing/gtest_util.h

```diff
@@ -358,6 +354,14 @@ void ArrayFromVector(const std::vector<C_TYPE>& values, std::shared_ptr<Array>*
   ArrayFromVector<TYPE, C_TYPE>(type, values, out);
 }
 
+template <typename TYPE, typename C_TYPE = typename TYPE::c_type>
+std::shared_ptr<Array> ArrayFromVector(std::vector<bool> is_valid,
+                                       std::vector<C_TYPE> values) {
+  std::shared_ptr<Array> out;
+  ArrayFromVector<TYPE, C_TYPE>(is_valid, values);
+  return out;
```

Comment:
Does this work? It looks like `out` isn't mutated in this function.

Reply:
Doesn't work; I started writing this overload and didn't end up using it. Will remove